doctrine/02-04-2021-DistributedOperatingSystemVoid.md

```diff
@@ -0,0 +1,225 @@
+---
+title: "The Distributed Operating System Void"
+date: "2021-04-02"
+author: "kris nóva"
+keywords: ""
+summary: ""
+---
+
+_Preface: I will be typing this in proper (capitalized) English. 
+Unlike the majority of my other work. 
+A substantial portion of this is IP is accredited to the concepts proposed by Andrew Rynhard and his work with Talos Systems. (COSi)_ 
+
+## Thesis
+
+Modern (circa 2021) UNIX based operating systems should be reimagined to support higher level workloads 
+that are managed in a distributed environment, such as kubernetes.
+
+The first step in progress towards this goal, is defining an interface.
+
+The Distributed Operating System Interface (`DOSi`) hopes to achieve this.
+
+# Problem Statement 
+
+Over the past 7 years Kubernetes has grown in popularity. 
+Kubernetes is built and managed on operating systems optimized for isolated workloads.
+Kubernetes mutates operating system level features, concerns, and services.
+Some of these services (CNI, Kubelet, Container Runtime, etc) are paramount for Kubernetes to run.
```

Comment:
I wonder if being Kubernetes-specific here and below is desired. We could have a statement, that it also applies to other distributed systems and Kubernetes is used more as an real-life example?

Reply:
while i agree that not being kubernetes specific is a good philosophy of the project (as we have it captured here https://github.com/cosi-project/community/pull/5/files#r614302069 in the charter)

this particular document is just a 1:1 copy of my blog https://nivenly.com/lib/2021-04-02-operating-system-interface/

and i *think it makes sense to keep the blog as literal as possible - part of the blog was intended to be very kubernetes specific to address the problem space concretely 

perhaps we can add some sort of meta information on this copy to show that its just a copy of the paper? or perhaps we can just reference the original copy and not add it to this PR?

i thought having a first concrete problem space write up here would be beneficial for the project as a hole :)

Reply:
>perhaps we can add some sort of meta information on this copy to show that its just a copy of the paper? or perhaps we can just reference the original copy and not add it to this PR?

I think it make sense to have it here as 1:1 copy (as backup, it's easier to find, grep for etc etc). Perhaps a note then, explaining that this applies to distributed system in general and that Kubernetes is used here as a specific example would be sufficient :+1: